app/javascript/Snackbar/Snackbar.jsx

```diff
@@ -1,15 +1,178 @@
-import { h } from 'preact';
+import { h, Component } from 'preact';
 import PropTypes from 'prop-types';
-import { snackbarItemProps } from './SnackbarItem';
+import { SnackbarItem } from './SnackbarItem';
 
-export const Snackbar = ({ children = [] }) => (
-  <div className={children.length > 0 ? 'crayons-snackbar' : 'hidden'}>
-    {children}
-  </div>
-);
+let snackbarItems = [];
 
-Snackbar.displayName = 'Snackbar';
+export function addSnackbarItem(snackbarItem) {
+  if (!Array.isArray(snackbarItem.actions)) {
+    snackbarItem.actions = []; // eslint-disable-line no-param-reassign
+  }
+
+  snackbarItems.push(snackbarItem);
+}
+
+export class Snackbar extends Component {
+  state = {
+    snacks: [],
+  };
+
+  pollingId;
+
+  paused = false;
+
+  pauseLifespan;
+
+  resumeLifespan;
+
+  componentDidMount() {
+    this.initializePolling();
+  }
+
+  componentDidUpdate() {
+    if (!this.pauseLifespan) {
+      this.pauseLifespan = (_event) => {
+        this.paused = true;
+      };
+
+      this.resumeLifespan = (event) => {
+        event.stopPropagation();
+        this.paused = false;
+      };
+
+      this.element.addEventListener('mouseover', this.pauseLifespan);
+      this.element.addEventListener('mouseout', this.resumeLifespan, true);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.element) {
+      this.element.removeEventListener('mouseover', this.pauseLifespan);
+      this.element.addEventListener('mouseout', this.resumeLifespan);
+    }
+  }
+
+  initializePolling() {
+    const { pollingTime, lifespan } = this.props;
+
+    this.pollingId = setInterval(() => {
+      if (snackbarItems.length > 0) {
+        // Need to add the lifespan to snackbar items because each second that goes by, we
+        // decrease the lifespan until it is no more.
+        const newSnacks = snackbarItems.map((snackbarItem) => ({
+          ...snackbarItem,
+          lifespan,
+        }));
+
+        snackbarItems = [];
+
+        this.updateSnackbarItems(newSnacks);
+
+        // Start the lifespan countdowns for each new snackbar item.
+        newSnacks.forEach((snack) => {
+          // eslint-disable-next-line no-param-reassign
+          snack.lifespanTimeoutId = setTimeout(() => {
+            this.decreaseLifespan(snack);
+          }, 1000);
+
+          if (snack.addCloseButton) {
+            // Adds an optional close button if addDefaultACtion is true.
+            snack.actions.push({
+              text: 'CLOSE',
```

Comment:
Lets not capitalize letters in action button - it always makes me feel like someone is shouting at me :D